### PR TITLE
chore: misc fixes to improve static typing and error detection

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR refactors various skill code to improve the static type checks and help us catch issues early.

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/base.py [hunk 1]
```diff
@@ -1600,7 +1599,6 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
             synchronized_data = self.synchronized_data.update(
                 synchronized_data_class=self.synchronized_data_class,
                 **{
-                    self.selection_key: frozenset(list(self.collection.keys())),
```
[the remark this answers is not in the document]
Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/base.py:1604]: This is a breaking change.

**Review by Adamantios — COMMENTED** [review]
Looks good but I have one concern shared in https://github.com/valory-xyz/open-autonomy/pull/1645#discussion_r1045898660.

File: packages/valory/skills/abstract_round_abci/test_tools/rounds.py [hunk 2]
```diff
@@ -139,9 +140,8 @@ def get_dummy_tx_payloads(
 class DummyRound(AbstractRound):
     """Dummy round."""
 
-    round_id = "round_id"
     allowed_tx_type = DummyTxPayload.transaction_type
-    payload_attribute = "value"
+    payload_attribute = get_name(DummyTxPayload.value)
```
Comment (0xArdi) [inline packages/valory/skills/abstract_round_abci/test_tools/rounds.py:144]: Just as a side note, it seems we provide the payload multiple times. Implicitly before, and explicitly now. We could refactor this in the future, to take the payload once, and infer these two class vars.

Comment (DavidMinarsch) [reply packages/valory/skills/abstract_round_abci/test_tools/rounds.py:144]: Yes that was the idea. That's why I am making all these links apparent so the refactor can be optimally done in the end.

File: packages/valory/skills/hello_world_abci/rounds.py [hunk 3]
```diff
@@ -130,6 +132,9 @@ class PrintMessageRound(CollectDifferentUntilAllRound, HelloWorldABCIAbstractRou
     allowed_tx_type = PrintMessagePayload.transaction_type
     payload_attribute = get_name(PrintMessagePayload.message)
 
+    # TODO: what's the point of this? It might be better
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply packages/valory/skills/hello_world_abci/rounds.py:135]: Ok, I'll add a ticket

[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply packages/valory/skills/hello_world_abci/rounds.py:135]: Addressed in https://github.com/valory-xyz/open-autonomy/pull/1686

**Pull request merged**